pkg/lib/ingress_gateway/ingress_gateway.go

```diff
@@ -0,0 +1,67 @@
+package ingressgateway
+
+import (
+	"context"
+	"encoding/json"
+	operatorv1alpha2 "github.com/kyma-project/istio/operator/api/v1alpha2"
+	"github.com/kyma-project/istio/operator/internal/filter"
+	"github.com/kyma-project/istio/operator/internal/reconciliations/istio"
```

Comment:
This lib now has a dependency on `internal/reconciliations/istio`, we should avoid this. I think we can also move all the lastAppliedConfiguration code out of the Istio package, because as far as I can see it's only used by restarters and istio_Controller.